lib/dependabot/shared_helpers.rb

```diff
@@ -62,35 +63,60 @@ def self.in_a_forked_process
     end
 
     class HelperSubprocessFailed < StandardError
-      def initialize(message, command)
+      def initialize(message, error_context:)
         super(message)
-        @command = command
+        @error_context = error_context
+        @command = error_context[:command]
       end
 
       def raven_context
-        { fingerprint: [@command] }
+        { fingerprint: [@command], extra: @error_context }
       end
     end
 
+    # rubocop:disable Metrics/MethodLength
     def self.run_helper_subprocess(command:, function:, args:, env: nil,
-                                   popen_opts: {})
-      raw_response = nil
-      popen_args = [env, command, "w+"].compact
-      IO.popen(*popen_args, popen_opts) do |process|
-        process.write(JSON.dump(function: function, args: args))
-        process.close_write
-        raw_response = process.read
+                                   std_err_to_std_out: false)
+      json_response = nil
+      stderr_output = nil
+      process_exit_value = nil
+
+      start = Time.now
+      Open3.popen3(*[env, command].compact) do |stdin, stdout, stderr, wait_thr|
+        stdin.write(JSON.dump(function: function, args: args))
+        stdin.close
+
+        stderr_output = stderr.read
+        json_response = stdout.read
+        json_response += stderr_output if std_err_to_std_out
+
+        process_exit_value = wait_thr.value
       end
+      time_taken = Time.now - start
 
-      response = JSON.parse(raw_response)
+      response = JSON.parse(json_response)
       return response["result"] if $CHILD_STATUS.success?
 
-      raise HelperSubprocessFailed.new(response["error"], command)
-    rescue JSON::ParserError
-      raise HelperSubprocessFailed.new(raw_response, command) if raw_response
+      error_context = {
+        command: command,
+        function: function,
+        args: args,
+        time_taken: time_taken,
+        stderr_output: stderr_output,
```

Comment:
I wonder if it's worth truncating this or removing the middle of the string if it's super long. I wonder what sentry would do if we gave it like 5mb of stderr. Might be fine!